tests/unit/wc-payment-api/test-class-wc-payments-api-client.php

```diff
@@ -103,6 +104,68 @@ public function test_create_intention_success() {
 		$this->assertEquals( $expected_status, $result->get_status() );
 	}
 
+
+	/**
+	 * Test a successful call to create_intention with fraud prevention enabled.
+	 *
+	 * @throws Exception - In the event of test failure.
+	 */
+	public function test_create_intention_with_fingerprinting_data() {
+		$expected_amount = 123;
+		$expected_status = 'succeeded';
+
+		$mock_fingerprinting   = $this->createMock( Buyer_Fingerprinting_Service::class );
+		$mock_fraud_prevention = $this->createMock( Fraud_Prevention_Service::class );
+
+		Buyer_Fingerprinting_Service::set_instance( $mock_fingerprinting );
+		Fraud_Prevention_Service::set_instance( $mock_fraud_prevention );
+
+		$mock_fraud_prevention
+			->expects( $this->exactly( 1 ) )
+			->method( 'is_enabled' )
+			->with()
+			->willReturn( true );
+
+		$mock_fingerprinting
+			->expects( $this->exactly( 1 ) )
+			->method( 'hash_data_for_fraud_prevention' )
+			->with( 1 );
+
+		$this->set_http_mock_response(
```

Comment:
It appears that this mock response and `assertSame` part below is irrelevant to the purpose of the test. Calling `create_intent` should be enough to verify that fingerprinting service has been called.